engine/access/rest/middleware/metrics.go

```diff
@@ -3,24 +3,20 @@ package middleware
 import (
 	"net/http"
 
-	"github.com/onflow/flow-go/module/metrics"
-
 	"github.com/slok/go-http-metrics/middleware"
 	"github.com/slok/go-http-metrics/middleware/std"
 
-	metricsProm "github.com/slok/go-http-metrics/metrics/prometheus"
-
 	"github.com/gorilla/mux"
-)
 
-func MetricsMiddleware() mux.MiddlewareFunc {
-	r := metrics.NewRestCollector(metricsProm.Config{Prefix: "access_rest_api"})
-	metricsMiddleware := middleware.New(middleware.Config{Recorder: r})
+	"github.com/onflow/flow-go/module"
+)
 
+func MetricsMiddleware(restCollector module.RestMetrics) mux.MiddlewareFunc {
+	metricsMiddleware := middleware.New(middleware.Config{Recorder: restCollector})
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 			// This is a custom metric being called on every http request
-			r.AddTotalRequests(req.Context(), req.Method, req.URL.Path)
+			restCollector.AddTotalRequests(req.Context(), req.Method, req.URL.Path)
```

Comment:
This seems to create a unique value of `handler` for every single collection, causing too. much metrics to be collected

<img width="714" alt="image" src="https://github.com/onflow/flow-go/assets/811374/0c65b697-0881-4270-ba98-cee69be16cae">